hadoop-hdds/framework/src/main/java/org/apache/hadoop/hdds/server/events/FixedThreadPoolWithAffinityExecutor.java

```diff
@@ -154,15 +175,77 @@ public long scheduledEvents() {
     return scheduled.value();
   }
 
+  @Override
+  public long droppedEvents() {
+    return dropped.value();
+  }
+
   @Override
   public void close() {
+    isRunning.set(false);
     for (ThreadPoolExecutor executor : executors) {
       executor.shutdown();
     }
+    EXECUTOR_MAP.clear();
   }
 
   @Override
   public String getName() {
     return name;
   }
+
+  /**
+   * Runnable class to perform execution of payload.
+   */
+  public static class ReportExecutor<P> implements Runnable {
```

Comment:
Actually it's a task. Can we rename it to something like "ContainerReportProcessTask"?